### PR TITLE
CIV-11028 Terms of agreement screen is not appearing

### PR DESCRIPTION
**Pull request by kalachandrasekar1**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-11028

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts [hunk 1]
```diff
@@ -166,3 +159,9 @@ function isPartialAdmissionPaidAndClaimantRejectPaymentOrNotSettleTheClaim(claim
 function isFullDefenceClaimantNotSettleTheClaim(claim: Claim) : boolean {
   return claim.claimantResponse?.hasFullDefenceStatesPaidClaimSettled?.option === YesNo.NO;
 }
+
+function isChooseHowFormaliseTask(claim: Claim) : boolean {
```
Comment (TatsyMaslova) [inline src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts:163]: as you can see this logic can be inside the claim. in addition it is not clear what it is doing or checking. perhaps rename the function?

File: src/test/unit/routes/guards/signSettlmentAgreementGuard.test.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ jest.mock('i18next', () => ({
 }));
 
 const CLAIM_ID = '123';
-const respondentIncompleteSubmissionUrl = constructResponseUrlWithIdParams(CLAIM_ID, RESPONSE_TASK_LIST_URL);
+const respondentIncompleteSubmissionUrl = constructResponseUrlWithIdParams(CLAIM_ID, CLAIMANT_RESPONSE_TASK_LIST_URL);
```
Comment (TatsyMaslova) [inline src/test/unit/routes/guards/signSettlmentAgreementGuard.test.ts:27]: wont this break R1? as the guard is now no longer applying for defendant response

File: src/main/common/models/claim.ts [hunk 3]
```diff
@@ -804,6 +804,11 @@ export class Claim {
   hasClaimantRejectedDefendantResponse() {
     return this?.claimantResponse?.hasFullDefenceStatesPaidClaimSettled?.option === YesNo.NO;
   }
+
+  isClaimantSignSettlementAgreement() {
```
[the remark this answers is not in the document]
Comment (kalachandrasekar1) [reply src/main/common/models/claim.ts:808]: Updated

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts [hunk 4]
```diff
@@ -67,16 +67,9 @@ export function buildWhatToDoNextSection(claim: Claim, claimId: string, lang: st
...
-        const proposeAlternativeRepayment = getProposeAlternativeRepaymentTask(claim, claimId, lang);
-        tasks.push(proposeAlternativeRepayment);
-
-        if ((claim.isPAPaymentOptionPayImmediately() && claim.claimantResponse?.courtProposedDate?.decision) ||
-          (claim.isPAPaymentOptionByDate() && claim.partialAdmission?.paymentIntention?.paymentDate)) {
-          const chooseHowFormaliseTask = getChooseHowFormaliseTask(claim, claimId, lang);
-          tasks.push(chooseHowFormaliseTask);
-        }
-
+      if (claim.claimantResponse?.fullAdmitSetDateAcceptPayment?.option === YesNo.YES && isChooseHowFormaliseTask(claim)) {
```
[the remark this answers is not in the document]
Comment (kalachandrasekar1) [reply src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts:70]: updated

File: src/test/unit/routes/guards/signSettlmentAgreementGuard.test.ts (hunk 2 above)
Comment (kalachandrasekar1) [reply src/test/unit/routes/guards/signSettlmentAgreementGuard.test.ts:27]: This will call only for claimant response

**Review by TatsyMaslova — APPROVED** [review]

**Review by jeswanth-hmcts — APPROVED** [review]

**Review by neeta-hmcts — APPROVED** [review]

**Pull request merged**